artifactory/commands/transferfiles/utils.go

```diff
@@ -338,17 +339,37 @@ func periodicallyUpdateThreadsAndStopStatus(pcWrapper *producerConsumerWrapper,
 			log.Debug("Stopping the polling on the settings and stop files for the current phase.")
 			return
 		}
-		if err := updateThreads(pcWrapper, buildInfoRepo); err != nil {
+
+		settings, err := utils.LoadTransferSettings()
+		if err != nil || settings == nil {
```

Comment:
We should not return if settings is `nil`, we should `continue`. Settings file does not exist on a clean run, until it is created with the settings command.

So far we also didn't return here if an error occurred. Our approach was to log it and not stop the transfer due to an error in the settings. If we decide to change approach, we should change it in all relevant places.

In addition, only log the error if it is not `nil`, this currently logs the error also when both the settings and error is `nil`.